tests/unit/models/gpflow/test_models.py

```diff
@@ -1547,19 +1557,19 @@ def test_sparse_variational_trajectory_sampler_has_correct_samples(
     x_predict = tf.constant([[1.0], [2.0], [3.0], [1.5], [2.5], [3.5]], gpflow.default_float())
     x_predict_parallel = tf.expand_dims(x_predict, -2)  # [N, 1, D]
     x_predict_parallel = tf.tile(x_predict_parallel, [1, num_samples, 1])  # [N, B, D]
-    samples = trajectory(x_predict_parallel)  # [N, B, 1]
-    sample_mean = tf.reduce_mean(samples, axis=1)  # [N, 1]
-    sample_variance = tf.math.reduce_variance(samples, axis=1)  # [N, 1]
+    samples = trajectory(x_predict_parallel)  # [N, B, P]
+    sample_mean = tf.reduce_mean(samples, axis=1)  # [N, P]
+    sample_variance = tf.math.reduce_variance(samples, axis=1)  # [N, P]
 
     true_mean, true_variance = model.predict(x_predict)
 
     # test predictions approx correct away from data
-    npt.assert_allclose(sample_mean[3:] + 1.0, true_mean[3:] + 1.0, rtol=0.1)
-    npt.assert_allclose(sample_variance[3:], true_variance[3:], rtol=0.25)
+    npt.assert_allclose(sample_mean[3:] + 1.0, true_mean[3:] + 1.0, rtol=0.12)
+    npt.assert_allclose(sample_variance[3:], true_variance[3:], rtol=0.3, atol=5e-3)
```

Comment:
Those tolerances are suspiciously high (in particular for the mean)... might be good to bump `num_samples` to 1000 or more if we can afford it? `rtol=0.3` away from the data worries me.

It is true that the trajectory sampler is an approximation of the exact sampler so we can't expect great accuracy but this still seems a bit high (which is also true for the previous version of this test). Also bump the number of features if possible?

Reply:
By increasing the number of samples to 3000 (without too much test time penalty), I have been able to reduce the `rtol` to `0.04` and `0.1` for mean and variance respectively (so a third compared to before). Is this sufficient?

Increasing the number of features increased the memory usage on the machine (as expected I guess), but didn't seem to help with the tolerances that much. Increasing it too large resulted in the process crashing due to memory issues. So I left it at the default of 1000.